Modules/Core/Common/include/itkMacro.h

```diff
@@ -99,18 +99,35 @@ namespace itk
 // to be quoted.
 #define ITK_PRAGMA(x) _Pragma(#x)
 
-// The clang compiler has many useful non-default compiler warnings
-// that tend to have a high false positive rate.
-// The following set of defines allows us to suppress false positives
-// and still track down suspicious code
-#if defined(__clang__) && defined(__has_warning)
-#  define CLANG_PRAGMA_PUSH ITK_PRAGMA(clang diagnostic push)
-#  define CLANG_PRAGMA_POP ITK_PRAGMA(clang diagnostic pop)
+// The GCC/Clang compilers have many useful non-default compiler warnings
+// that tend to have a high false positive rate or are otherwise not always appropriate.
+// The following set of defines allows us to suppress instances of said warnings.
+
+// For GCC and Clang (Clang also identifies itself as GCC, and supports these pragmas):
+#if defined(__GNUC__) && defined(__has_warning)
```

Comment:
defined(__has_warning) is only defined on clang.  GCC 13.2 does not have a __has_warning method.

Reply:
Oh damn, I thought they had added it too a while ago.

Well, as it happens both those warnings have been around forever, so don't really need conditionalization.